tests/Backend/CommonPart1/CreateTableWithConfigurationTest.php

```diff
@@ -0,0 +1,367 @@
+
+        $this->configId = sha1($this->generateDescriptionForTestObject());
+
+        // delete configuration for this test
+        try {
+            $this->componentsClient->deleteConfiguration(self::COMPONENT_ID, $this->configId);
+        } catch (ClientException $e) {
+            if (preg_match('/Configuration cannot be deleted because it is being used in following configured tables: (.*). Delete them first./', $e->getMessage(), $out)) {
+                $tablesToDelete = explode(', ', $out[1]);
+                foreach ($tablesToDelete as $tableId) {
+                    $this->client->dropTable($tableId);
+                }
+                $this->componentsClient->deleteConfiguration(self::COMPONENT_ID, $this->configId);
+            } elseif (preg_match('/Configuration \w+ not found/', $e->getMessage(), $out)) {
+                // noop, config already deleted
+            } else {
+                // throw other
+                throw $e;
+            }
+        }
+    }
+
+    protected function assertMetadata(array $table): void
+    {
+        $expected = [
+            'ID' => [
+                'KBC.datatype.type' => 'NUMBER',
+                'KBC.datatype.nullable' => '1',
+                'KBC.datatype.basetype' => 'NUMERIC',
+                'KBC.datatype.length' => '38,0',
+            ],
+            'NAME' => [
+                'KBC.datatype.type' => 'VARCHAR',
+                'KBC.datatype.nullable' => '1',
+                'KBC.datatype.basetype' => 'STRING',
+                'KBC.datatype.length' => '100',
+            ],
+        ];
+        $actual = [];
+        foreach ($table['columnMetadata'] as $columnName => $metadatum) {
+            $actual[$columnName] = [];
+            foreach ($metadatum as $item) {
+                $actual[$columnName][$item['key']] = $item['value'];
+            }
+        }
+        $this->assertEquals($expected, $actual);
+    }
+
+    public function testTableCreate(): void
+    {
+        // create test configuration
+        $configuration = (new Configuration())
+            ->setComponentId(self::COMPONENT_ID)
+            ->setConfigurationId($this->configId)
+            ->setName($this->configId)
+            ->setConfiguration([
+                'migrations' => [
+                    [
+                        'sql' => 'CREATE TABLE {{ id(bucketName) }}.{{ id(tableName) }} (id integer, name varchar(100))',
+                        'description' => 'first ever',
+                    ],
+                ],
+            ]);
+        $this->componentsClient->addConfiguration($configuration);
+
+        // create table from config
+        $tableName = 'custom-table-1';
+        $configurationOptions = (new TableWithConfigurationOptions($tableName, $this->configId));
+        $tableId = $this->_client->createTableWithConfiguration(
+            $this->getTestBucketId(),
+            $configurationOptions
+        );
+
```

Comment:
tady by to mělo otestovat dvě eventy - jednak `storage.tableCreated` a jednak `storage.tableWithConfigurationMigrated` bych řekl... teď nám navíc ta tableCreated eventa chybí úplně :(

Reply:
Testy na eventy jsou pridany.